frontend/src/stimulus/controllers/dynamic/storages/storage-audience.controller.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * -- copyright
+ * OpenProject is an open source project management software.
+ * Copyright (C) the OpenProject GmbH
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License version 3.
+ *
+ * OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+ * Copyright (C) 2006-2013 Jean-Philippe Lang
+ * Copyright (C) 2010-2013 the ChiliProject Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * See COPYRIGHT and LICENSE files for more details.
+ * ++
+ */
+
+import { Controller } from '@hotwired/stimulus';
+
+export default class StorageAudienceController extends Controller {
+  static targets = [
+    'audienceInput',
+    'audienceInputWrapper',
+    'idpRadio',
+  ];
+
+  declare readonly audienceInputTarget:HTMLInputElement;
+  declare readonly audienceInputWrapperTarget:HTMLDivElement;
+  declare readonly idpRadioTarget:HTMLInputElement;
+
+  connect() {
+    if (this.idpRadioTarget.checked) {
+      this.hideAudienceInput();
+    }
+  }
+
+  showAudienceInput() {
+    this.audienceInputTarget.value = '';
```

Comment:
:yellow_circle: 
Setting value to be empty on every show leads to user loosing its input which can be inconvenient. For example, in `Automatically managed folder` section of the same page password is not removed when user clicks on checkbox.
Therefore I suggest not clearing input value, but then `__op-idp__` can appear in the input field. I'm not sure if that is critical.

Reply:
> but then __op-idp__ can appear in the input field

This is exactly why I added the clearing, to prevent this "magic" audience from leaking through.

I understand the general inconvenience of losing an input, on the other hand we could at best prevent losing it when you click around in the form. As soon as you save, your input is lost anyways, because the radio button is just a way to set the same data.

Let me try something...

Reply:
Thanks for nudging me over. I found a solution that is much cleaner, because we never even send the internal identifier to the browser.

Feel free to have a look at my latest commit.

Reply:
Looks nice to me! Thanks for taking care of it.